src/core/tsi/s2a/handshaker/s2a_handshaker_client.cc

```diff
@@ -259,14 +283,16 @@ static void S2AHandshakerCallUnref(void* arg, grpc_error* error) {
   grpc_call_unref(call);
 }
 
-S2AHandshakerClient::~S2AHandshakerClient() {
-  if (gpr_unref(&refs_)) {
+void S2AHandshakerClient::Unref() {
+  if (gpr_unref(refs_)) {
     if (call_ != nullptr) {
       grpc_core::ExecCtx::Run(DEBUG_LOCATION,
                               GRPC_CLOSURE_CREATE(S2AHandshakerCallUnref, call_,
                                                   grpc_schedule_on_exec_ctx),
                               GRPC_ERROR_NONE);
     }
+    gpr_free(refs_);
+    refs_ = nullptr;
```

Comment:
Why do we set it `nullptr` after freeing it?

Reply:
It is not needed. Good catch! I've removed it in the next commit.